Dockerfile

```diff
@@ -1,28 +1,60 @@
 FROM php:8.2-apache
+RUN useradd nonroot
 LABEL maintainer="Mark Tompsett <mark_tompsett@sil.org>"
 
 ENV REFRESHED_AT 2023-07-12
 
 # Make sure apt has current list/updates
+USER root
 RUN apt-get update -y \
 # Fix timezone stuff from hanging.
     && echo "America/New_York" > /etc/timezone \
-    && apt-get install -y tzdata \
+    && apt-get install -y --no-install-recommends tzdata \
     && apt-get upgrade -y \
 # Install
-    && apt-get install -y \
+    && apt-get install -y --no-install-recommends \
 # things needed for GoogleMock objects
         sqlite3 \
 # some basics
         unzip wget zip \
 # Clean up to reduce docker image size
     && apt-get clean \
-    && rm -rf /var/lib/apt/lists/*
+    && rm -rf /var/lib/apt/lists/* \
+# Make sure the directory we'll mount and reference is there
+    && mkdir -p /data
 
-RUN mkdir -p /data
+USER nonroot
 WORKDIR /data
-COPY ./ /data
+COPY actions-services.yml /data
+COPY composer-install.sh /data
+COPY composer.json /data
+COPY composer.lock /data
+COPY docker-compose.yml /data
+COPY Dockerfile /data
+COPY LICENSE /data
+COPY Makefile /data
+COPY README.md /data
+COPY run-tests.sh /data
+COPY .travis.yml /data
+COPY SilMock/ /data/SilMock
 
-RUN cd /data && ./composer-install.sh
-RUN mv /data/composer.phar /usr/bin/composer
+USER root
+# Make sure the development test files exist and have writable permissions
+RUN touch /data/SilMock/DataStore/Sqlite/Test1_Google_Service_Data.db && \
+    touch /data/SilMock/DataStore/Sqlite/Test2_Google_Service_Data.db && \
+    touch /data/SilMock/DataStore/Sqlite/Test3_Google_Service_Data.db && \
+    touch /data/SilMock/DataStore/Sqlite/Test4_Google_Service_Data.db && \
+    touch /data/SilMock/DataStore/Sqlite/Test5_Google_Service_Data.db && \
+    touch /data/SilMock/tests/.phpunit.result.cache && \
+    chown -R nonroot:root /data && \
+    chmod 664 /data/SilMock/DataStore/Sqlite/Test1_Google_Service_Data.db && \
+    chmod 664 /data/SilMock/DataStore/Sqlite/Test2_Google_Service_Data.db && \
+    chmod 664 /data/SilMock/DataStore/Sqlite/Test3_Google_Service_Data.db && \
+    chmod 664 /data/SilMock/DataStore/Sqlite/Test4_Google_Service_Data.db && \
+    chmod 664 /data/SilMock/DataStore/Sqlite/Test5_Google_Service_Data.db && \
+    chmod 664 /data/SilMock/tests/.phpunit.result.cache
+
+WORKDIR /data
+RUN ./composer-install.sh && mv /data/composer.phar /usr/bin/composer
```

Comment:
If at some point you want to simplify this (to possibly get rid of the need for a `composer-install.sh` script), you can also just copy the composer binary from their Docker image, like so:
```
COPY --from=composer:2 /usr/bin/composer /usr/local/bin/composer
```
You can use [whatever version tag](https://hub.docker.com/_/composer/tags) you want to, after the colon in that `--from` parameter.